File: credit/src/main/java/ru/patterns/credit/infrastructure/messaging/sender/KafkaMetricSender.java
package ru.patterns.credit.infrastructure.messaging.sender;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Component;
import ru.patterns.credit.infrastructure.messaging.KafkaAsyncSender;
import ru.patterns.credit.shared.kafka.MetricDto;

@Component
public class KafkaMetricSender extends KafkaAsyncSender<MetricDto> {

    public KafkaMetricSender(KafkaTemplate<String, Object> template,
                             ObjectMapper objectMapper) {
        super(template, objectMapper, "metrics-topic");
    }
}

File: core/src/main/kotlin/ru/patterns/core/utils/metric/KafkaMetricSender.kt
package ru.patterns.core.utils.metric

import com.fasterxml.jackson.databind.ObjectMapper
import org.springframework.stereotype.Component
import reactor.kafka.sender.KafkaSender
import ru.patterns.core.utils.KafkaAsyncSender
import ru.patterns.core.utils.metric.serialization.MetricDto

@Component
class KafkaMetricSender(
    kafkaSender: KafkaSender<String, String>,
    objectMapper: ObjectMapper,
) : KafkaAsyncSender<MetricDto>(
    kafkaSender = kafkaSender,
    objectMapper = objectMapper,
    topic = "metrics-topic"
)